src/qibo/transpiler/blocks.py

```diff
@@ -65,8 +65,8 @@ def fuse(self, block, name: Optional[str] = None):
             (:class:`qibo.transpiler.blocks.Block`): fusion of the two input blocks.
         """
         if not self.qubits == block.qubits:
-            raise_error(
-                BlockingError, "In order to fuse two blocks their qubits must coincide."
+            raise BlockingError(
+                "In order to fuse two blocks their qubits must coincide."
```

Comment:
Should let `raise_error` deal with the exception since it's a `qibo` exception and not a regular `python` exception

Reply:
I modified this line because if I use the qibo raise_error the error message is printed several times when i call the block decomposition function:
```python
while len(initial_blocks) > 0:
        first_block = initial_blocks[0]
        remove_list = [first_block]
        if len(initial_blocks[1:]) > 0:
            for second_block in initial_blocks[1:]:
                try:
                    first_block = first_block.fuse(second_block)
                    remove_list.append(second_block)
                except BlockingError:
                    if not first_block.commute(second_block):
                        break
        blocks.append(first_block)
        _remove_gates(initial_blocks, remove_list)
``` 
The error message is printed in the try/except. 
Maybe a good solution would be to directly change the raire_error function

Reply:
I think first we should find out why it's being raised multiple times

Reply:
```python
def raise_error(exception, message=None, args=None):
    """Raise exception with logging error.

    Args:
        exception (Exception): python exception.
        message (str): the error message.
    """
    log.error(message)
    if args:
        raise exception(message, args)
    else:
        raise exception(message)
``` 
i think the log.error(message) is the problem